pkg/aws_cloudwatch.go

```diff
@@ -307,19 +307,24 @@ func getFilteredMetricDatas(region string, accountId *string, namespace string,
 		if len(dimensionNameList) > 0 && !metricDimensionsMatchNames(cwMetric, dimensionNameList) {
 			continue
 		}
-		for _, dimension := range cwMetric.Dimensions {
-			if dimensionFilterValues, ok := dimensionsFilter[*dimension.Name]; ok {
-				if d, ok := dimensionFilterValues[*dimension.Value]; !ok {
+
+		for _, dimension := range cwMetric.Dimensions { // loop over the returned dimensions for our current metric
```

Comment:
Comments are a bit overwhelming here, maybe you could write just 1-2 lines _before_ the loop to explain what the code is intended to achieve at high level?

Reply:
done!